src/Tickets/Commerce/Communication/Email.php

```diff
@@ -66,4 +72,18 @@ public function send_tickets_email( $order_id, $post_id = null ) {
 		// Send the emails.
 		tribe( Module::class )->send_tickets_email_for_attendees( $to_send, $send_args );
 	}
+
+	/**
+	 * Update email sent counter for the attendee.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $attendee_id Attendee ID.
+	 *
+	 * @return bool
+	 */
+	public function update_ticket_sent_counter( int $attendee_id ): bool {
+		$prev_val = (int) get_post_meta( $attendee_id, Attendee::$ticket_sent_meta_key, true );
+		return update_post_meta( $attendee_id, Attendee::$ticket_sent_meta_key, $prev_val + 1 );
```

Comment:
Do not return this value.
The `update_post_meta` function will return `false` if the **new** value is the same as the **previous** value; so its return value indicates whether it operated or not, but tells you nothing about whether the desired status (i.e. the current value of the meta field after the update operation is the previous + 1).
These two lines of code are **not** atomic; hence the function is thread-safe (i.e. multiple PHP threads running the same code might change it at the same time).

If you need to return a boolean value from this function, return one that indicates the updated value is the expected one:

```php
return get_post_meta(...) === $prev_val + 1;
```

This way you're saying the function is a success if, at the end, the value in the database is the expected one, however that came to be.